src/Microsoft.SqlTools.ServiceLayer/DacFx/DacFxOperation.cs

```diff
@@ -35,15 +39,18 @@ abstract class DacFxOperation : ITaskOperation
 
         protected DacServices DacServices { get; private set; }
 
+        protected bool TelemetryEnabled { get; private set; }
+
         protected ConnectionInfo ConnInfo { get; private set; }
 
-        protected DacFxOperation(ConnectionInfo connInfo)
+        protected DacFxOperation(ConnectionInfo connInfo, bool telemetryEnabled = false)
```

Comment:
Instead of passing it in why not just update the library directly when the config changes? Right now you won't change the telemetry state until you start the next operation which isn't really what we want.

Reply:
hmm yes you are right and I'm wondering now why it was working ...

Reply:
does a static field work?

Reply:
Not really I think - you need each operation itself to be updated separately since they're each creating their own services instance. So instead I think you'd have DacFxService watch for the changes and then loop through each operation and tell it to update the telemetry enabled state for each one.

Reply:
I already added the watch in the DacFxService class. and that service doesn't have access the the operations. operations are created for new requests